rp2040-hal/src/i2c.rs

```diff
@@ -23,6 +23,14 @@ use rp2040_pac::{I2C0, I2C1, RESETS};
 pub enum Error {
     /// I2C abort with error
     Abort(u32),
+    /// User passed in a read buffer that was 0 or >255 length
+    InvalidReadBufferLength(u32),
+    /// User passed in a write buffer that was 0 or >255 length
+    InvalidWriteBufferLength(u32),
+    /// Target i2c address is out of range
+    AddressOutOfRange(u32),
+    /// Target i2c address is reserved
+    AddressReserved(u32),
```

Comment:
u32 should be u8 here, as that's the size we use for an I2C address.

Reply:
Yep, true. I was quite tired when I implemented this, thought I should keep the types the same for some reason.
I've updated it to be u8 now.